### ISSUE TITLE
llvm backend: llvm assert hit when compiling for mips and passing vector for f80 to a function

### ISSUE THREAD

**Issue by antlilja** [ziglang/zig#16479]
### Zig Version

0.11.0-dev.4059+17255bed4

### Steps to Reproduce and Observed Behavior

The compiler hits an LLVM assert when passing a f80 vector to a function and compiling for mips

```zig test test.zig -target mips64-linux-none``` (with a zig compiler linked to a debug version of LLVM)

```zig
// test.zig

fn foo(vec: @Vector(1, f80)) void {
    _ = vec;
}

test "test" {
    var x = foo(@Vector(1, f80){0.0});
    _ = x;
}
```

```
llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:767: getCopyToPartsVector: Assertion `Val.getValueType() == BuiltVectorTy && "Unexpected vector value type"' failed.
```

It seems to be and issue with LLVM itself and has already been reported in the [LLVM issue tracker](https://github.com/llvm/llvm-project/issues/63608). But I thought it wouldn't hurt to file an issue if anyone else runs into it.

### Expected Behavior

Does not crash

**Comment by andrewrk** [comment]
Thanks for reporting the issue upstream!

**Comment by tgross35** [comment]
Is this closed since https://github.com/llvm/llvm-project/commit/e49103b2790f346b436b280a2da7a1c6a9f2ab3d was applied?

**Comment by Rexicon226** [comment]
> Is this closed since [llvm/llvm-project@e49103b](https://github.com/llvm/llvm-project/commit/e49103b2790f346b436b280a2da7a1c6a9f2ab3d) was applied?

No, because this was applied to a version of LLVM that we haven't upgraded to yet. The error still happens in master branch.

**Comment by Rexicon226** [comment]
> that commit made it into llvm 17 so if it fixed the issue it would already be fixed on zig master. so it seems there's more investigating to be done.

You're right, I missread the date of the commit 😄 

The current error I'm getting on master is the same one antilja reported. However the reduced case reported in the LLVM issue passes for me now. So it seems there is another edge case here of some sort.

**Comment by alexrp** [comment]
Still happens with LLVM 19:

```
/home/alexrp/Source/llvm-project/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:811: void getCopyToPartsVector(llvm::SelectionDAG&, const llvm::SDLoc&, llvm::SDValue, llvm::SDValue*, unsigned int, llvm::MVT, const llvm::Value*, std::optional<unsigned int>): Assertion `Val.getValueType() == BuiltVectorTy && "Unexpected vector value type"' failed.
```
